cfd_protocol/src/lib.rs

```diff
@@ -28,12 +30,27 @@ const MIN_RELAY_FEE: u64 = 1;
 const P2PKH_DUST_LIMIT: u64 = 546;
 
 pub trait WalletExt {
-    fn build_lock_psbt(&self, amount: Amount, identity_pk: PublicKey) -> PartyParams;
+    // TODO: the naming of this fn feels a bid odd
+    fn build_party_params(&self, amount: Amount, identity_pk: PublicKey) -> Result<PartyParams>;
 }
 
-impl<B, D> WalletExt for bdk::Wallet<B, D> {
-    fn build_lock_psbt(&self, amount: Amount, identity_pk: PublicKey) -> PartyParams {
-        todo!()
+impl<B, D> WalletExt for bdk::Wallet<B, D>
+where
+    D: BatchDatabase,
+{
+    fn build_party_params(&self, amount: Amount, identity_pk: PublicKey) -> Result<PartyParams> {
+        let mut builder = self.build_tx();
+        builder
+            .ordering(bdk::wallet::tx_builder::TxOrdering::Bip69Lexicographic)
+            .add_recipient(Script::new(), amount.as_sat());
```

Comment:
Can we have a comment here why it is okay to pass an empty script? Does the size of the script here affect fee calculation?

Reply:
This could be a source of errors as we are underpaying for the fees. 
@luckysori : I think we didn't notice this because I added  `.fee_rate(FeeRate::from_sat_per_vb(2.0))`. 

Did we run the docker-test with `1.0 sat/vbyte` as well?

Reply:
Can confirm, this is an issue if fees are set to 1.0
` RpcError { code: -26, message: "min relay fee not met, 238 < 252", data: None }`

Reply:
Fixed.